python/DEVELOPMENT.md

```diff
@@ -16,36 +16,41 @@
 
 ### Linux and macOS
 
-First, set up your thirdparty C++ toolchain using libraries from conda-forge:
+#### System Requirements
+
+On macOS, any modern XCode (6.4 or higher; the current version is 8.3.1) is
+sufficient.
+
+On Linux, for this guide, we recommend using gcc 4.8 or 4.9, or clang 3.7 or
+higher. You can check your version by running
 
 ```shell
-conda config --add channels conda-forge
+$ gcc --version
+```
 
-export ARROW_BUILD_TYPE=Release
+On Ubuntu 16.04 and higher, you can obtain gcc 4.9 with:
 
-export CPP_TOOLCHAIN=$HOME/cpp-toolchain
-export LD_LIBRARY_PATH=$CPP_TOOLCHAIN/lib:$LD_LIBRARY_PATH
+```shell
+$ sudo apt-get install g++-4.9
+```
 
-export BOOST_ROOT=$CPP_TOOLCHAIN
-export FLATBUFFERS_HOME=$CPP_TOOLCHAIN
-export RAPIDJSON_HOME=$CPP_TOOLCHAIN
-export THRIFT_HOME=$CPP_TOOLCHAIN
-export ZLIB_HOME=$CPP_TOOLCHAIN
-export SNAPPY_HOME=$CPP_TOOLCHAIN
-export BROTLI_HOME=$CPP_TOOLCHAIN
-export JEMALLOC_HOME=$CPP_TOOLCHAIN
-export ARROW_HOME=$CPP_TOOLCHAIN
-export PARQUET_HOME=$CPP_TOOLCHAIN
+Finally, set gcc 4.9 as the active compiler using:
 
-conda create -y -q -p $CPP_TOOLCHAIN \
-      flatbuffers rapidjson boost-cpp thrift-cpp snappy zlib brotli jemalloc
+```shell
+export CC=gcc-4.9
```

Comment:
The output of $(which FOO) doesn't produce an executable command for me when FOO is found multiple times in the path. I'll leave this as is for now

```
$ which gcc
gcc is /home/wesm/.conda/envs/$NAME/bin/gcc
gcc is /usr/bin/gcc
```